Implement Entity Framework 6 (EF6) auto instrumentation (#560)
elastic/apm-agent-dotnet: docs/setup.asciidoc

* https://www.nuget.org/packages/Elastic.Apm.AspNetCore[Elastic.Apm.AspNetCore]: This package contains ASP.NET Core monitoring related code. The main difference between this package and the `Elastic.Apm.NetCoreAll` package is that this package does not reference the `Elastic.Apm.EntityFrameworkCore` package, so if you have an ASP.NET Core application that does not use EF Core and you want to avoid adding additional unused references, you should use this package.
* https://www.nuget.org/packages/Elastic.Apm.EntityFrameworkCore[Elastic.Apm.EntityFrameworkCore]: This package contains EF Core monitoring related code.
* https://www.nuget.org/packages/Elastic.Apm.AspNetFullFramework[Elastic.Apm.AspNetFullFramework]: This package contains ASP.NET (Full .NET Framework) monitoring related code.

> * https://www.nuget.org/packages/Elastic.Apm.EntityFramework6[Elastic.Apm.EntityFramework6]: This package contains interceptor that automatically creates spans for DB operations executed by Entity Framework 6 (EF6) on behalf of the application.

Reviewer: ```suggestion
* https://www.nuget.org/packages/Elastic.Apm.EntityFramework6[Elastic.Apm.EntityFramework6]: This package contains an interceptor that automatically creates spans for DB operations executed by Entity Framework 6 (EF6) on behalf of the application.
```

----

You can also configure the agent using `web.config` as described at <<configuration-on-asp-net>>.

[float]
[[setup-ef6]]
=== Entity Framework 6

You can enable auto instrumentation for Entity Framework 6 by referencing the `Elastic.Apm.EntityFramework6` package
and including `Ef6Interceptor` interceptor in your application's `web.config`:

[source,xml]
----
<?xml version="1.0" encoding="utf-8"?>
<configuration>
    <entityFramework>
        <interceptors>
            <interceptor type="Elastic.Apm.EntityFramework6.Ef6Interceptor, Elastic.Apm.EntityFramework6" />
        </interceptors>		
    </entityFramework>
</configuration>
----

As an alternative to registering the interceptor via the configuration, you can register it in the application code:   
[source,csharp]
----
DbInterception.Add(new Elastic.Apm.EntityFramework6.Ef6Interceptor());
----

> For example, if it's ASP.NET MVC application you can place the above call in `Application_Start` method.

Reviewer: ```suggestion
For example, in an ASP.NET MVC application, you can place the above call in the `Application_Start` method.
```
